src/foam/nanos/http/SessionWebAgent.java

```diff
@@ -83,13 +83,14 @@ public void execute(X x) {
       // check permissions
       Subject subject = new Subject.Builder(x).setUser(user).build();
       session.setContext(session.getContext().put("subject", subject));
-      if ( ! auth.check(session.getContext(), permission_) ) {
+      var applyX = session.applyTo(x);
```

Comment:
with  applyTo the flow also does not need the explicit subject setup and session.setContext (subject). so lines 84, and 85.